Tests/ClientRuntimeTests/OrchestratorTests/OrchestratorTests.swift

```diff
@@ -0,0 +1,1251 @@
+                "modifyBeforeRetryLoop",
+                "readBeforeAttempt",
+                "selectAuthScheme",
+                "applyEndpoint",
+                "modifyBeforeSigning",
+                "readBeforeSigning",
+                "applySigner",
+                "readAfterSigning",
+                "modifyBeforeTransmit",
+                "readBeforeTransmit",
+                "executeRequest",
+                "readAfterTransmit",
+                "modifyBeforeDeserialization",
+                "readBeforeDeserialization",
+                "deserialize",
+                "readAfterDeserialization",
+                "modifyBeforeAttemptCompletion",
+                "readAfterAttempt",
+                "errorInfo",
+                "modifyBeforeCompletion",
+                "readAfterExecution",
+            ]
+        )
+    }
+
+    func test_stepErrorHandlingModifyBeforeCompletion() async {
+        let modifyBeforeCompletionTrace = Trace()
+        let modifyBeforeCompletion = await asyncResult {
+            let b = self.traceOrchestrator(trace: modifyBeforeCompletionTrace)
+            b.interceptors.addModifyBeforeCompletion({ _ in throw TestError(value: "1") })
+            b.interceptors.addModifyBeforeCompletion({ _ in throw TestError(value: "2") })
+            return try await b.build().execute(input: TestInput(foo: ""))
+        }
+
+        XCTAssertEqual(modifyBeforeCompletion, .failure(TestError(value: "2")))
+        XCTAssertEqual(
+            modifyBeforeCompletionTrace.trace,
+            [
+                "readBeforeExecution",
+                "modifyBeforeSerialization",
+                "readBeforeSerialization",
+                "serialize",
+                "readAfterSerialization",
+                "modifyBeforeRetryLoop",
+                "readBeforeAttempt",
+                "selectAuthScheme",
+                "applyEndpoint",
+                "modifyBeforeSigning",
+                "readBeforeSigning",
+                "applySigner",
+                "readAfterSigning",
+                "modifyBeforeTransmit",
+                "readBeforeTransmit",
+                "executeRequest",
+                "readAfterTransmit",
+                "modifyBeforeDeserialization",
+                "readBeforeDeserialization",
+                "deserialize",
+                "readAfterDeserialization",
+                "modifyBeforeAttemptCompletion",
+                "readAfterAttempt",
+                "modifyBeforeCompletion",
+                "readAfterExecution",
+            ]
+        )
+    }
+
+    func test_stepErrorHandlingReadAfterExecution() async {
+        let readAfterExecutionTrace = Trace()
+        let readAfterExecution = await asyncResult {
+            let b = self.traceOrchestrator(trace: readAfterExecutionTrace)
+            b.interceptors.addReadAfterExecution({ _ in throw TestError(value: "1") })
+            b.interceptors.addReadAfterExecution({ _ in throw TestError(value: "2") })
```

Comment:
For these tests (there are a number of them), what exactly are these testing? I see `"2"` is being asserted, but not `"1"`.

Reply:
These are making sure the most recent error thrown from an interceptor is what is returned